src/applications/vaos/services/patient/index.js

```diff
@@ -0,0 +1,388 @@
+/**
+ * Functions related to patient specific information
+ * @module services/Patient
+ */
+import environment from 'platform/utilities/environment';
+import {
+  checkPastVisits,
+  getLongTermAppointmentHistory,
+  getRequestLimits,
+} from '../var';
+import { recordEligibilityFailure, recordVaosError } from '../../utils/events';
+import { captureError } from '../../utils/error';
+import { ELIGIBILITY_REASONS } from '../../utils/constants';
+import { promiseAllFromObject } from '../../utils/data';
+import { getAvailableHealthcareServices } from '../healthcare-service';
+
+/**
+ * @typedef PatientEligibilityForType
+ * @global
+ *
+ * @property {boolean} hasRequiredAppointmentHistory Has had appointment in the past that meets VATS requirements
+ *   - Mapped from past visits check
+ * @property {?boolean} isEligibleForNewAppointmentRequest Is under the request limit
+ *   - Mapped from request limits check
+ */
+
+/**
+ * @typedef PatientEligibility
+ * @global
+ *
+ * @property {?PatientEligibilityForType} direct Patient eligibility for direct scheduling
+ * @property {?PatientEligibilityForType} request Patient eligibility for requests
+ */
+
+/**
+ * @typedef {'error'|'overRequestLimit'|'noEnabled'|'notSupported'|'noRecentVisit'|
+ *   'noClinics'|'noMatchingClinics'} EligibilityReason
+ * @global
+ */
+
+/**
+ * @typedef FlowEligibility
+ * @global
+ *
+ * @property {boolean} direct Can the patient use the direct schedule flow
+ * @property {EligibilityReason} directReason The reason the patient isn't eligible for direct flow
+ * @property {boolean} request Can the patient use the request flow
+ * @property {EligibilityReason} requestReason The reason the patient isn't eligible for request flow
+ */
+
+function createErrorHandler(errorKey) {
+  return data => {
+    captureError(data, true);
+    recordVaosError(`eligibility-${errorKey}`);
+    return new Error('Eligibility error');
+  };
+}
+
+const PRIMARY_CARE = '323';
+
+const DISABLED_LIMIT_VALUE = 0;
+
+function hasVisitedInPastMonths(pastVisit) {
+  if (!pastVisit) {
+    return true;
+  }
+
+  return (
+    pastVisit.durationInMonths === DISABLED_LIMIT_VALUE ||
+    pastVisit.hasVisitedInPastMonths
+  );
+}
+
+function isUnderRequestLimit(data) {
+  return (
+    data?.requestLimit === DISABLED_LIMIT_VALUE ||
+    data?.numberOfRequests < data?.requestLimit
+  );
+}
+
+async function fetchPatientEligibilityFromVAR({
```

Comment:
This contains the logic for call the different past visit and request limit endpoints in VAR. None of this logic is necessary with the VAOS service, so it's called from another function, where we would have a feature flag toggle.